frontends/election-manager/src/hooks/use_election_manager_store.ts

```diff
@@ -0,0 +1,697 @@
+    });
+
+    return isOfficialResults;
+  }, [logger, storage]);
+
+  const getIsOfficialResultsQuery = useQuery<boolean>(
+    [isOfficialResultsKey],
+    async () => {
+      return (await loadIsOfficialResults()) ?? false;
+    }
+  );
+  const isOfficialResults = getIsOfficialResultsQuery.data;
+
+  const setStorageKeyAndLog = useCallback(
+    async (storageKey: string, value: unknown, logDescription: string) => {
+      try {
+        await storage.set(storageKey, value);
+        await logger.log(LogEventId.SaveToStorage, currentUserRoleRef.current, {
+          message: `${logDescription} successfully saved to storage.`,
+          storageKey,
+          disposition: 'success',
+        });
+      } catch (error) {
+        assert(error instanceof Error);
+        await logger.log(LogEventId.SaveToStorage, currentUserRoleRef.current, {
+          message: `Failed to save ${logDescription} to storage.`,
+          storageKey,
+          error: error.message,
+          disposition: 'failure',
+        });
+      }
+    },
+    [logger, storage]
+  );
+
+  const removeStorageKeyAndLog = useCallback(
+    async (storageKey: string, logDescription: string) => {
+      try {
+        await storage.remove(storageKey);
+        await logger.log(LogEventId.SaveToStorage, currentUserRoleRef.current, {
+          message: `${logDescription} successfully cleared in storage.`,
+          storageKey,
+          disposition: 'success',
+        });
+      } catch (error) {
+        assert(error instanceof Error);
+        await logger.log(LogEventId.SaveToStorage, currentUserRoleRef.current, {
+          message: `Failed to clear ${logDescription} in storage.`,
+          storageKey,
+          error: error.message,
+          disposition: 'failure',
+        });
+      }
+    },
+    [logger, storage]
+  );
+
+  const reset = useCallback(async () => {
+    const previousElection = electionDefinition;
+    if (previousElection) {
+      void logger.log(
+        LogEventId.ElectionUnconfigured,
+        currentUserRoleRef.current,
+        {
+          disposition: LogDispositionStandardTypes.Success,
+          previousElectionHash: previousElection.electionHash,
+        }
+      );
+    }
+    // we set a new election definition, reset everything
+    try {
+      await storage.clear();
+      await logger.log(LogEventId.SaveToStorage, currentUserRoleRef.current, {
+        message:
+          'All current data in storage, including election definition, cast vote records, tallies, and printed ballot information cleared.',
+        disposition: 'success',
+      });
+    } catch (error) {
+      assert(error instanceof Error);
+      await logger.log(LogEventId.SaveToStorage, currentUserRoleRef.current, {
+        message: 'Failed clearing all current data in storage.',
+        disposition: 'failure',
+        error: error.message,
+      });
+    }
+    void queryClient.invalidateQueries([electionDefinitionStorageKey]);
```

Comment:
I don't remember `queryClient.invalidateQueries` being async from my previous experience with react-query. Are you sure the types are working?

Reply:
From the [docs](https://tanstack.com/query/v4/docs/reference/QueryClient#queryclientinvalidatequeries):

```tsx
await queryClient.invalidateQueries(['posts'], {
  exact,
  refetchType: 'active',
}, { throwOnError, cancelRefetch })
```

Reply:
TIL!

I was always following the examples from the [guide](https://tanstack.com/query/v4/docs/guides/invalidations-from-mutations) which don't away the promise...